collector/mdadm_linux.go

```diff
@@ -140,9 +142,9 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 
 	lines := strings.Split(mdStatusFile, "\n")
 	var (
-		currentMD           string
-		personality         string
-		active, total, size int64
```

Comment:
Instead of carrying around a bunch of variables for each of the `mdStatus` struct fields, we can just remove all of these variables here and build a new `mdStatus` type on the fly. This would have also prevented the bug you fixed in the other PR.

```go
// Replace `currentMD = mainLine[0]` with:
md := mdStatus{
    name:  mainLine[0],
    active: mainLine[2] == "active",
}

// and then in all the other lines like `failed = int64(strings.Count(l, "(F)"))`
md.disksFailed = int64(strings.Count(l, "(F)"))
```